# 
# Submitted:
#
# Robert Schwebel, 2004-03-29
#
# Error: 
#
# Does not compile because linux fixes seem to be out of date.  
#
# Description: 
#
# see Error 
# 
# State:
# 
# Fixed in local tree, not sent upstream yet. 
#

diff -urN bing-1.0.5/bing.c bing-1.0.5-ptx/bing.c
--- bing-1.0.5/bing.c	2001-01-19 19:28:53.000000000 +0100
+++ bing-1.0.5-ptx/bing.c	2004-03-29 17:28:07.000000000 +0200
@@ -671,13 +671,8 @@
 		pr_retip((struct ip *)ICMP_TO_DATA(icp));
 		break;
 	case ICMP_PARAMPROB:
-#ifdef linux
-		(void)printf("Parameter problem: pointer = 0x%02x\n",
-		    icp->un.gateway);
-#else
 		(void)printf("Parameter problem: pointer = 0x%02x\n",
 		    icp->icmp_hun.ih_pptr);
-#endif /* linux */
 		pr_retip((struct ip *)ICMP_TO_DATA(icp));
 		break;
 	case ICMP_TSTAMP:
diff -urN bing-1.0.5/sysdep.h bing-1.0.5-ptx/sysdep.h
--- bing-1.0.5/sysdep.h	1995-07-18 00:36:44.000000000 +0200
+++ bing-1.0.5-ptx/sysdep.h	2004-03-29 17:28:22.000000000 +0200
@@ -46,58 +46,6 @@
 #ifndef _SYSDEP_H_
 #define _SYSDEP_H_
 
-#ifdef linux
-#define icmp		icmphdr
-#define ip		iphdr
-#define icmp_type	type
-#define icmp_code	code
-#define icmp_cksum	checksum
-#define icmp_id		un.echo.id
-#define icmp_seq	un.echo.sequence
-#define icmp_gwaddr	un.gateway
-#define ip_hl		ihl
-#define ip_v		version
-#define ip_tos		tos
-#define ip_len		tot_len
-#define ip_id		id
-#define ip_off		frag_off
-#define ip_ttl		ttl
-#define ip_p		protocol
-#define ip_sum		check
-#define ip_src		saddr
-#define ip_dst		daddr
-#define MAX_IPOPTLEN	4096
-#define ICMP_MINLEN	16
-
-#define IPOPT_MINOFF	4
-
-#define IPOPT_EOL	IPOPT_END
-#define IPOPT_NOP	IPOPT_NOOP
-
-#define ICMP_UNREACH		ICMP_DEST_UNREACH
-#define ICMP_UNREACH_NET	ICMP_NET_UNREACH
-#define ICMP_UNREACH_HOST	ICMP_HOST_UNREACH
-#define ICMP_UNREACH_PROTOCOL	ICMP_PROT_UNREACH
-#define ICMP_UNREACH_PORT	ICMP_PORT_UNREACH
-#define ICMP_UNREACH_NEEDFRAG	ICMP_FRAG_NEEDED
-#define ICMP_UNREACH_SRCFAIL	ICMP_SR_FAILED
-#define ICMP_SOURCEQUENCH	ICMP_SOURCE_QUENCH
-#define ICMP_REDIRECT_NET	ICMP_REDIR_NET
-#define ICMP_REDIRECT_HOST	ICMP_REDIR_HOST
-#define ICMP_REDIRECT_TOSNET	ICMP_REDIR_NETTOS
-#define ICMP_REDIRECT_TOSHOST	ICMP_REDIR_HOSTTOS
-#define ICMP_TIMXCEED		ICMP_TIME_EXCEEDED
-#define ICMP_TIMXCEED_INTRANS	ICMP_EXC_TTL
-#define ICMP_TIMXCEED_REASS	ICMP_EXC_FRAGTIME
-#define ICMP_PARAMPROB		ICMP_PARAMETERPROB
-#define ICMP_TSTAMP		ICMP_TIMESTAMP
-#define ICMP_TSTAMPREPLY	ICMP_TIMESTAMPREPLY
-#define ICMP_IREQ		ICMP_INFO_REQUEST
-#define ICMP_IREQREPLY		ICMP_INFO_REPLY
-
-#define	ICMP_TO_DATA(icp)	((u_char *)(icp + 1))
-#endif /* linux */
-
 /* Defaults (should work on most systems) */
 
 #ifndef ICMP_TO_DATA
